File: Striver's SDE Sheet/PascalsTriangle/code.py
def generateRows(row):
    res=1
    ans=[]
    ans.append(res)

    for i in range(1,row):
        res=res*(row-i)//i;
        ans.append(res)

    return ans

def generate(numRows):
    ans=[]

    for i in range(1,numRows+1):
        ans.append(generateRows(i))
    
    return ans

numRows=5
ans=generate(numRows)

for i in ans:
    print(" ".join(map(str,i)))

File: Striver's SDE Sheet/PascalsTriangle/code.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int>generateRows(int row){
    vector<int> ans;
    int res=1;
    ans.push_back(res);

    for(int i=1;i<row;i++){
        res=res*(row-i)/i;
        ans.push_back(res);
    }
    return ans;
}

vector<vector<int>> generate(int numRows){
    vector<vector<int>> ans;

    for(int i=1;i<=numRows;i++){
        ans.push_back(generateRows(i));
    }
    return ans;
}

int main(){
    int numRows=5;
    vector<vector<int>>ans=generate(numRows);

    for(auto ele:ans){
        for(int num:ele){
            cout<<num<<" ";
        }
        cout<<endl;
    }
}